src/nft/components/NFTCard.tsx

```diff
@@ -0,0 +1,68 @@
+import { useCallback } from 'react';
+import { useAccount } from 'wagmi';
+import { border, cn, color, pressable } from '../../styles/theme';
+import { useIsMounted } from '../../useIsMounted';
+import { useTheme } from '../../useTheme';
+import { LifecycleType, type NFTCardReact } from '../types';
+import { NFTLifecycleProvider } from './NFTLifecycleProvider';
+import { NFTProvider } from './NFTProvider';
+
+export function NFTCard({
+  children,
+  className,
+  contractAddress,
+  tokenId,
+  useNFTData,
+  onStatus,
+  onError,
+  onSuccess,
+}: NFTCardReact) {
+  const componentTheme = useTheme();
+
+  const isMounted = useIsMounted();
+
+  const { chain } = useAccount();
+
+  const handleOnClick = useCallback(() => {
+    const network = chain?.name.toLowerCase() ?? 'base';
+    const openSeaUrl = `https://opensea.io/assets/${network}/${contractAddress}/${tokenId}`;
+    window.open(openSeaUrl, '_blank', 'noopener,noreferrer');
+  }, [chain, contractAddress, tokenId]);
+
+  // prevents SSR hydration issue
+  if (!isMounted) {
+    return null;
+  }
```

Comment:
sorry if i missed this in the other PR but did we decide that we needed this `isMounted` return in both `NFTCard` and `NFTMintCard` ?

Reply:
Originally I had the same styling in both so I could extract the divs and these hooks to a common component, but, that was incorrect, both these cards are pretty different so I need to handle them separately.